### PR TITLE
Ensure that the root element exist before rendering CustomerEffortScoreTrac…

### LINKED ISSUE #48110: TypeError T is null
### Prerequisites

- [X] I have carried out troubleshooting steps and I believe I have found a bug.
- [X] I have searched for similar bugs in both open and closed issues and cannot find a duplicate.

### Describe the bug

Uncaught TypeError: t is null
    <anonymous> http://server.tst/wp-content/plugins/woocommerce/assets/client/admin/app/index.js?ver=1717421290:2
    <anonymous> http://server.tst/wp-content/plugins/woocommerce/assets/client/admin/app/index.js?ver=1717421290:2
    <anonymous> http://server.tst/wp-content/plugins/woocommerce/assets/client/admin/app/index.js?ver=1717421290:2
    <anonymous> http://server.tst/wp-content/plugins/woocommerce/assets/client/admin/app/index.js?ver=1717421290:2

### Expected behavior

I don't write the js code

### Actual behavior

No error please.

### Steps to reproduce

1) I am using the latest wordpress trunk version
2) Woocommerce beta 9.0.0-beta.2 
3) i have the calssic editor plugin active
3) i edit a page and look into the browser js console.  The error is there as described above,

### WordPress Environment

<details><summary>System Status Report</summary>

```
### WordPress Environment ###

WordPress address (URL): http://server.tst
Site address (URL): http://server.tst
WC Version: 9.0.0-beta.2
Legacy REST API Package Version: The Legacy REST API plugin is not installed on this site.
Action Scheduler Version: ✔ 3.7.4
Log Directory Writable: ✔
WP Version: 6.6-alpha-58044
WP Multisite: –
WP Memory Limit: 512 MB
WP Debug Mode: ✔
WP Cron: ✔
Language: en_US
External object cache: –

### Server Environment ###

Server Info: Apache/2.4.52 (Ubuntu)
PHP Version: 8.3.7
PHP Post Max Size: 512 MB
PHP Time Limit: 600
PHP Max Input Vars: 5000
cURL Version: 7.81.0
OpenSSL/3.0.2

SUHOSIN Installed: –
MySQL Version: 8.0.36-0ubuntu0.22.04.1
Max Upload Size: 512 MB
Default Timezone is UTC: ✔
fsockopen/cURL: ✔
SoapClient: ❌ Your server does not have the SoapClient class enabled - some gateway plugins which use SOAP may not work as expected.
DOMDocument: ✔
GZip: ✔
Multibyte String: ✔
Remote Post: ✔
Remote Get: ✔

### Database ###

WC Database Version: 8.6.1
WC Database Prefix: wp_
Total Database Size: 16.13MB
Database Data Size: 12.70MB
Database Index Size: 3.43MB
wp_woocommerce_sessions: Data: 0.13MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_api_keys: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_attribute_taxonomies: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_downloadable_product_permissions: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_woocommerce_order_items: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_order_itemmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_tax_rates: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_woocommerce_tax_rate_locations: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_shipping_zones: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_woocommerce_shipping_zone_locations: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
wp_woocommerce_shipping_zone_methods: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_woocommerce_payment_tokens: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_payment_tokenmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_log: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_actionscheduler_actions: Data: 0.05MB + Index: 0.11MB + Engine InnoDB
wp_actionscheduler_claims: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_actionscheduler_groups: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_actionscheduler_logs: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_commentmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_comments: Data: 0.02MB + Index: 0.09MB + Engine InnoDB
wp_e_events: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_e_notes: Data: 0.02MB + Index: 0.17MB + Engine InnoDB
wp_e_notes_users_relations: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
wp_e_submissions: Data: 0.02MB + Index: 0.27MB + Engine InnoDB
wp_e_submissions_actions_log: Data: 0.02MB + Index: 0.11MB + Engine InnoDB
wp_e_submissions_values: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_eg_grids: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_eg_item_elements: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_eg_item_skins: Data: 0.08MB + Index: 0.03MB + Engine InnoDB
wp_eg_navigation_skins: Data: 0.14MB + Index: 0.02MB + Engine InnoDB
wp_fbv: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_fbv_attachment_folder: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_gla_attribute_mapping_rules: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_gla_budget_recommendations: Data: 0.19MB + Index: 0.11MB + Engine InnoDB
wp_gla_merchant_issues: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_gla_shipping_rates: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_gla_shipping_times: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_links: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_options: Data: 7.13MB + Index: 0.08MB + Engine InnoDB
wp_postmeta: Data: 1.52MB + Index: 0.53MB + Engine InnoDB
wp_posts: Data: 1.52MB + Index: 0.13MB + Engine InnoDB
wp_responsive_menu_pro: Data: 0.06MB + Index: 0.00MB + Engine InnoDB
wp_revslider_css: Data: 0.13MB + Index: 0.02MB + Engine InnoDB
wp_revslider_css_bkp: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_revslider_layer_animations: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_revslider_layer_animations_bkp: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_revslider_navigations: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_revslider_navigations_bkp: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_revslider_sliders: Data: 0.05MB + Index: 0.02MB + Engine InnoDB
wp_revslider_sliders7: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_revslider_sliders_bkp: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_revslider_slides: Data: 0.13MB + Index: 0.02MB + Engine InnoDB
wp_revslider_slides7: Data: 0.09MB + Index: 0.02MB + Engine InnoDB
wp_revslider_slides_bkp: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_revslider_static_slides: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_revslider_static_slides_bkp: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_term_relationships: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_term_taxonomy: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_termmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_terms: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_usermeta: Data: 0.05MB + Index: 0.03MB + Engine InnoDB
wp_users: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
wp_wc_admin_note_actions: Data: 0.05MB + Index: 0.02MB + Engine InnoDB
wp_wc_admin_notes: Data: 0.05MB + Index: 0.00MB + Engine InnoDB
wp_wc_category_lookup: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wc_customer_lookup: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_download_log: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_order_addresses: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_wc_order_coupon_lookup: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_order_operational_data: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_order_product_lookup: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_wc_order_stats: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
wp_wc_order_tax_lookup: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_orders: Data: 0.02MB + Index: 0.11MB + Engine InnoDB
wp_wc_orders_meta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_product_attributes_lookup: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wc_product_download_directories: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wc_product_meta_lookup: Data: 0.02MB + Index: 0.09MB + Engine InnoDB
wp_wc_rate_limits: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wc_reserved_stock: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wc_tax_rate_classes: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wc_webhooks: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wpomp_group_map: Data: 0.00MB + Index: 0.00MB + Engine MyISAM
wp_wpomp_locations: Data: 0.00MB + Index: 0.00MB + Engine MyISAM
wp_wpomp_maps: Data: 0.01MB + Index: 0.00MB + Engine MyISAM

### Post Type Counts ###

attachment: 222
elementor_library: 1
nav_menu_item: 105
page: 120
portfolio: 29
post: 19
preset: 8
product: 27
product_variation: 7
revision: 57
shop_order_placehold: 5
slideshow: 5
wp_global_styles: 1
wp_navigation: 1

### Security ###

Secure connection (HTTPS): ❌
					Your store is not using HTTPS. Learn more about HTTPS and SSL Certificates.
Hide errors from visitors: ❌Error messages should not be shown to visitors.

### Active Plugins (3) ###

Classic Editor: by WordPress Contributors – 1.6.3
WooCommerce Beta Tester: by WooCommerce – 2.3.1
WooCommerce: by Automattic – 9.0.0-beta.2

### Inactive Plugins (62) ###

Akismet Anti-spam: Spam Protection: by Automattic - Anti-spam Team – 5.3.2
All in One SEO: by All in One SEO Team – 4.6.3
AVIF Support [GrandPlugins]: by GrandPlugins – 1.0.8
bbPress: by The bbPress Contributors – 2.6.9
Champion Responsive Menu: by The ThemeBuilders – 3.1.47
Contact Form 7: by Takayuki Miyoshi – 5.9.5
Design Plus Responsive Menu: by The Themebuilders – 3.1.46
Disable Author Pages: by Frank Staude – 0.11
Duplicator: by Duplicator – 1.5.9
Elementor: by Elementor.com – 3.21.8
Elementor Pro: by Elementor.com – 3.21.2
Element Pack Pro: by BdThemes – 7.12.10
ElementsKit Lite: by Wpmet – 3.1.4
ElementsKit Pro: by Wpmet – 3.6.2
Enable jQuery Migrate Helper: by The WordPress Team – 1.4.0
Envato Theme Check: by Scott Parry – 1.0.0
Essential Grid: by ThemePunch – 3.1.2.2
Essential Grid Google Fonts Addon: by ThemePunch – 1.1.3
Essential Grid Media Filters Addon: by ThemePunch – 1.0.2
Essential Grid NGG Adapter AddOn: by ThemePunch – 1.0.3
Essential Grid Right Click Protection AddOn: by ThemePunch – 1.0.2
Essential Grid RML Adapter AddOn: by ThemePunch – 1.0.3
Essential Grid Social Media AddOn: by ThemePunch – 1.0.6
Essential Grid Video Playlists AddOn: by ThemePunch – 1.0.5
Essential Grid Watermarks Addon: by ThemePunch – 1.0.3
Essential Grid WP Gallery Replace Addon: by ThemePunch – 1.0.3
Fancy Product Designer: by fancyproductdesigner.com – 4.6.7
Fancy Product Designer: by fancyproductdesigner.com – 4.6.9
Fancy Product Designer Plus: by fancyproductdesigner.com – 1.3.0
Fancy Product Designer REST API: by fancyproductdesigner.com – 1.6.4
FileBird Lite: by Ninja Team – 6.2.3
FileBird Pro: by Ninja Team – 6.1.2
Flexible Elementor Panel: by WebMat – 2.4.0
Google Analytics for WordPress by MonsterInsights: by MonsterInsights – 8.27.0
Gravity Forms: by Gravity Forms – 2.8.4
Gutenberg: by Gutenberg Team – 18.4.1
Hello Dolly: by Matt Mullenweg – 1.7.2
Justified Image Grid: by Firsh – 4.5
Mega Main Menu: by MegaMain.com – 2.2.2
OoohBoi Steroids for Elementor: by OoohBoi – 2.1.6
PRO Elements: by PROElements.org – 3.18.3
QuadMenu: by QuadLayers – 3.1.2
QuadMenu PRO: by Mega Menu – 1.9.0
Quform: by ThemeCatcher – 2.14.2
Replace Google Fonts with Bunny Fonts: by antonioleutsch – 2.1.2
Responsive Menu: by ExpressTech – 4.3.5
Reveal IDs: by Oliver Schlöbe – 1.6.1
RTL Tester: by <a href="http://blog.yoavfarhi.com">Yoav Farhi</a>
<a href="http://automattic.com">Automattic</a> – 1.2

Slider Revolution: by ThemePunch – 6.7.12
Striking_R Framework Extension: by kaptinlin – 2.3.4
Super Elements for Elementor: by ThemeRevealed – 2.5.2
Widget Importer & Exporter: by ChurchThemes.com – 1.6.1
WooCommerce Multilingual & Multicurrency: by OnTheGoSystems – 5.3.6
WordPress Beta Tester: by WordPress Upgrade/Install Team – 3.5.5
WordPress Importer: by wordpressdotorg – 0.8.2
WP-PageNavi: by Lester 'GaMerZ' Chan – 2.94.1
WP-Sweep: by Lester 'GaMerZ' Chan – 1.1.8
WP Leaflet Maps Pro: by flippercode – 1.1.0
WPML Media: by OnTheGoSystems – 2.7.3
WPML Multilingual CMS: by OnTheGoSystems – 4.6.3
WPML String Translation: by OnTheGoSystems – 3.2.5
Yoast SEO: by Team Yoast – 22.8

### Must Use Plugins (1) ###

Xcelerated Media Library Custom Fields: by xceleratedonline – 1.0.0

### Settings ###

API Enabled: –
Force SSL: –
Currency: EUR (€)
Currency Position: left_space
Thousand Separator: .
Decimal Separator: ,
Number of Decimals: 2
Taxonomies: Product Types: external (external)
grouped (grouped)
simple (simple)
variable (variable)

Taxonomies: Product Visibility: exclude-from-catalog (exclude-from-catalog)
exclude-from-search (exclude-from-search)
featured (featured)
outofstock (outofstock)
rated-1 (rated-1)
rated-2 (rated-2)
rated-3 (rated-3)
rated-4 (rated-4)
rated-5 (rated-5)

Connected to WooCommerce.com: –
Enforce Approved Product Download Directories: ✔
HPOS feature enabled: ✔
Order datastore: Automattic\WooCommerce\Internal\DataStores\Orders\OrdersTableDataStore
HPOS data sync enabled: –

### Logging ###

Enabled: ✔
Handler: Automattic\WooCommerce\Internal\Admin\Logging\LogHandlerFileV2
Retention period: 30 days
Level threshold: –
Log directory size: 230 KB

### WC Pages ###

Shop base: #5903 - /shop/
Cart: #5904 - /cart/
Checkout: #5905 - /checkout/
My account: #5906 - /my-account/
Terms and conditions: ❌ Page not set

### Theme ###

Name: Twenty Twenty-Four
Version: 1.1
Author URL: https://wordpress.org
Child Theme: ❌ – If you are modifying WooCommerce on a parent theme that you did not build personally we recommend using a child theme. See: How to create a child theme
WooCommerce Support: ❌ Not declared

### Templates ###

Overrides: /var/docs/server.tst/public/wp-content/plugins/woocommerce/templates/block-notices/error.php
/var/docs/server.tst/public/wp-content/plugins/woocommerce/templates/block-notices/notice.php
/var/docs/server.tst/public/wp-content/plugins/woocommerce/templates/block-notices/success.php


### Admin ###

Enabled Features: activity-panels
analytics
product-block-editor
coupons
core-profiler
customize-store
import-products-task
experimental-fashion-sample-products
shipping-smart-defaults
shipping-setting-tour
homescreen
marketing
mobile-app-banner
navigation
onboarding
onboarding-tasks
product-custom-fields
remote-inbox-notifications
remote-free-extensions
payment-gateway-suggestions
shipping-label-banner
subscriptions
store-alerts
transient-notices
woo-mobile-welcome
wc-pay-promotion
wc-pay-welcome-page
launch-your-store

Disabled Features: minified-js
new-product-management-experience
pattern-toolkit-full-composability
product-pre-publish-modal
settings
async-product-editor-category-field
beta-tester-slotfill-examples

Daily Cron: ✔ Next scheduled: 2024-02-26 10:21:36 +00:00
Options: ✔
Notes: 71
Onboarding: completed

### Action Scheduler ###

Complete: 1
Oldest: 2024-04-30 11:32:29 +0000
Newest: 2024-04-30 11:32:29 +0000

Pending: 54
Oldest: 2024-02-26 10:21:38 +0000
Newest: 2024-05-14 13:42:15 +0000

Canceled: 4
Oldest: -0001-11-30 00:00:00 +0000
Newest: -0001-11-30 00:00:00 +0000


### Status report information ###

Generated at: 2024-06-03 13:34:23 +00:00
```
</details>

### Isolating the problem

- [X] I have deactivated other plugins and confirmed this bug occurs when only WooCommerce plugin is active.
- [X] This bug happens with a default WordPress theme active, or [Storefront](https://woocommerce.com/products/storefront/).
- [X] I can reproduce this bug consistently using the steps above.

### PR DESCRIPTION
**Pull request by moon0326**
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

Closes #48110 .

This PR ensures that the root element exists before attempting to render the CustomerEffortScoreTracksContainer component.

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Create a new JN site with this branch and WooCommerce Beta Tester.
2. Open the browser inspector
3. Go to Posts -> Click on any posts.
4. Confirm no `TypeError` occurs

### Changelog entry

-   [x] Automatically create a changelog entry from the details below.

-   [ ] This Pull Request does not require a changelog entry. (Comment required below)

<details>

<summary>Changelog Entry Details</summary>

#### Significance

-   [x] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [x] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

Ensures that the root element exists before attempting to render the CustomerEffortScoreTracksContainer component.

#### Comment 

</details>

### CODE REVIEW

File: plugins/woocommerce/changelog/48199-fix-make-sure-element-exist-for-customerscore [hunk 1]
```diff
@@ -0,0 +1,4 @@
+Significance: patch
+Type: fix
+
+</details>  <details>  <summary>Changelog Entry Comment</summary>  Ensures that the root element exists before attempting to render the CustomerEffortScoreTracksContainer component.
```
Comment (chihsuan) [inline plugins/woocommerce/changelog/48199-fix-make-sure-element-exist-for-customerscore:4]:
```suggestion
Ensures that the root element exists before attempting to render the CustomerEffortScoreTracksContainer component.
```

**Review by chihsuan — APPROVED** [review]
Thanks for the quick fix! Tested well and LGTM.

**Review by ilyasfoo — APPROVED** [review]
Tests well, LGTM! 🚢

**Comment by rjchow** [comment]
@chihsuan / @ilyasfoo were you able to get the CES survey to show up? It's not this PR because I can't get it to show up on trunk either, but it seems to work for 8.9.2

File: plugins/woocommerce/changelog/48199-fix-make-sure-element-exist-for-customerscore (hunk 1 above)
Comment (moon0326) [reply plugins/woocommerce/changelog/48199-fix-make-sure-element-exist-for-customerscore:4]: Thanks @chihsuan again 👍 

I'm keep making the same mistake 😞

**Comment by chihsuan** [comment]
> were you able to get the CES survey to show up? It's not this PR because I can't get it to show up on trunk either, but it seems to work for 8.9.2

 Good catch! I just tried it and can't get it to show up either.

**Comment by moon0326** [comment]
Closing in favor of https://github.com/woocommerce/woocommerce/pull/48235, which addresses the same issue.

**Pull request closed without merging**